tests/testthat/test-nest.R

```diff
@@ -49,3 +49,13 @@ test_that("nesting works for empty data frames", {
   out <- nest(df, x, y)
   expect_equal(out$data, list(df))
 })
+
+test_that("tibble conversion occurs in the `nest.data.frame()` method", {
+  tbl <- mtcars %>% nest(-am, -cyl)
+  expect_s3_class(tbl, "tbl_df")
+  expect_true(
```

Comment:
I think this is probably slightly too much — I'd recommend just checking the value of the first element of the list (i.e. `tbl$data[[1]]`)

Reply:
Sure no problem ... though it seems this change has induced some downstream unit test failures elsewhere. I'll look into it some more and see if I can come up with something better.

Reply:
It looks like unconditionally converting to tibble class causes the nesting variable, e.g. `x`, to remain in the returned object once `nest()` is called. It should be dropped. This is causing all the downstream issues. I'll do some digging, but I was hoping this one was going to be relatively simple :(

Reply:
Here is the issue ... the `as_tibble()` ungroups the tibble if it's already a tibble.
```r
df <- tibble(x = c(1, 1, 1), y = 1:3) %>% dplyr::group_by(x)
df %>% dplyr::is_grouped_df()     # TRUE   
df %>% as_tibble() %>% dplyr::is_grouped_df().  # FALSE
```
This might be difficult to work around without digging into `as_tibble()` (aside, the new tibble pkg `v2.0.1` has been causing me headaches), but coercing the output seems like a bit of a hack? Maybe?
Stripping the `grouped_df` class causes a bunch of downstream things *not* to happen, like getting the `group_vars` etc. This seems a bit like a `tibble::as_tibble()` issue :(

Perhaps the condition is the simplest way forward? At least until `as_tibble()` doesn't strip the `grouped_df` class. What do you suggest?

Reply:
Maybe we need another `tbl.df` method that doesn't do the coercion? So `nest.data.frame()` would coerce to a tibble, and then call `nest.tbl_df()`?  (Let me know if that's not enough info to know what to do next)

Reply:
I'd put it above it, but yeah.

Reply:
Hold on ... I just re-read what you said and realize I misunderstood. There's a lot of duplicated code in my solution, I'll just call `nest.tbl_df()` rather than repeat the action part of the function. Can you squash the commits on your end?